pkg/networkservice/common/heal/client.go

```diff
@@ -114,22 +157,23 @@ func (f *healClient) recvEvent() {
 			case networkservice.ConnectionEventType_DELETE:
 				for _, conn := range event.GetConnections() {
 					delete(f.reported, conn.GetId())
-					if f.requestors[conn.GetId()] != nil {
-						f.requestors[conn.GetId()]()
-					}
 				}
 			}
+
 			for id, request := range f.requestors {
 				if _, ok := f.reported[id]; !ok {
 					request()
 				}
 			}
 		})
 	}
+
 	f.recvEventExecutor.AsyncExec(f.recvEvent)
 }
 
 func (f *healClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	f.lazyInit(request.GetConnection().GetId())
```

Comment:
I'm confused, why are we calling lazyInit on a per connection id basis?  And why is lazyInit using SyncExec and thus causing every Request to effectively be processed one by one instead of in parallel?

Reply:
Fixed with eliminating of laziness